libs/libc/tls/Kconfig

```diff
@@ -45,10 +45,11 @@ config TLS_LOG2_MAXSTACK
 
 config TLS_NELEM
 	int "Number of TLS elements"
-	default 0
+	default 4
```

Comment:
This means that TLS (and pthread-specific) is always enabled? (for defconfigs that don't set it to 0 as above)
BTW, the CONFIG_NPTHREAD_KEYS is not used anymore it should be removed from sched/Kconfig.

Reply:
Yes, this was for backward compatibility.  When TLS and pthread-specific data were merged, the the old CONFIG_NPTHREAD_KEYS was replaced with CONFIG_TLS_ELEM since TLS is now the underlying mechanism.  In order to be backward compatible with existing defconfig files, CONFIG_TLS_NELEM must have the same default as CONFIG_PTHREAD_KEYS used to have, that is four.

In most defconfig files, pthread-specific data is disabled like:

    -CONFIG_NPTHREAD_KEYS=0
    +CONFIG_TLS_NELEM=0

So there will be no impact to the the behavior of any of the configurations.  So although this looks weird, this is the correct default to have absolutely no impact on the existing configurations.